tasks-around-university/android_app/components/main_game/Scanner.js

```diff
@@ -0,0 +1,116 @@
+import React, { Component } from 'react';
+import {
+  Alert,
+  Linking,
+  Dimensions,
+  LayoutAnimation,
+  Text,
+  View,
+  StatusBar,
+  StyleSheet,
+  TouchableOpacity,
+} from 'react-native';
+import ScannerStyles from './styles/ScannerStyles'
+
+
+import Layout from '../testscan/Layout';
+import { BarCodeScanner, Permissions } from 'expo';
+
+export default class Scanner extends Component {
+  state = {
+    hasCameraPermission: null,
+    lastScannedUrl: null,
+  };
+
+  componentDidMount() {
+    this._requestCameraPermission();
+  }
+
+  _requestCameraPermission = async () => {
+    const { status } = await Permissions.askAsync(Permissions.CAMERA);
+    this.setState({
+      hasCameraPermission: status === 'granted',
+    });
+  };
+
+  _handleBarCodeRead = result => {
+    if (result.data !== this.state.lastScannedUrl) {
+      LayoutAnimation.spring();
+      this.setState({ lastScannedUrl: result.data });
+    }
+  };
+  render() {
+    return (
+      <View style={ScannerStyles.container}>
+        {this.state.hasCameraPermission === null
+          ? <Text>Requesting for camera permission</Text>
+          : this.state.hasCameraPermission === false
+              ? <Text style={{ color: '#fff' }}>
+                  Camera permission is not granted
+                </Text>
+              : <BarCodeScanner
+                  onBarCodeRead={this._handleBarCodeRead}
+                  style={[StyleSheet.absoluteFill]
+                  }/>}
+                  <View style={ScannerStyles.topOverlay} />
+                  <View style={ScannerStyles.leftOverlay} />
+                  <View style={ScannerStyles.rightOverlay} />
+                  <View style={ScannerStyles.bottomOverlay} />
+                  <View style={ScannerStyles.topLeftCorner} />
+                  <View style={ScannerStyles.topRightCorner} />
+                  <View style={ScannerStyles.bottomLeftCorner} />
+                  <View style={ScannerStyles.bottomRightCorner} />
+                  <View style={ScannerStyles.header}>
+                  <Text style={ScannerStyles.headerText}>Scan QR Code to start</Text>
+                  </View>
+               
```

Comment:
<View style={styles.footer}>
                    <TouchableOpacity
                      onPress={this._handlePressCancel1}
                      hitSlop={{ top: 40, bottom: 40, right: 40, left: 40 }}>
                      <Text style={styles.cancelText}>Cancel</Text>
                    </TouchableOpacity>
                  </View>
täst viel toi cancel button qr ruutuun. Kutsuu sen handlePresscancel1.